crates/gosub_css3/src/tokenizer.rs

```diff
@@ -906,37 +896,29 @@ impl<'stream> Tokenizer<'stream> {
     }
 
     pub fn tell(&self) -> usize {
-        self.cur_location.offset() as usize
+        self.stream.offset()
     }
 
-    pub fn slice(&self, start: usize, end: usize) -> String {
+    pub fn slice(&mut self, start: usize, end: usize) -> String {
+        let old_pos = self.stream.offset();
+        self.stream.seek(start);
+
+        // todo: this is not efficient
         let mut s = String::new();
```

Comment:
You could use `String::with_capacity(end - start)` (or maybe even only do the calculation once (or maybe the compiler does that?))

Reply:
fixed